src/coreComponents/physicsSolvers/multiphysics/CompositionalMultiphaseReservoirAndWells.cpp

```diff
@@ -0,0 +1,406 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file CompositionalMultiphaseReservoirAndWells.cpp
+ *
+ */
+
+#include "CompositionalMultiphaseReservoirAndWells.hpp"
+
+#include "common/TimingMacros.hpp"
+#include "constitutive/fluid/MultiFluidBase.hpp"
+#include "mesh/PerforationExtrinsicData.hpp"
+#include "physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.hpp"
+#include "physicsSolvers/fluidFlow/CompositionalMultiphaseHybridFVM.hpp"
+#include "physicsSolvers/fluidFlow/CompositionalMultiphaseUtilities.hpp"
+#include "physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWell.hpp"
+#include "physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWellExtrinsicData.hpp"
+#include "physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWellKernels.hpp"
+#include "physicsSolvers/fluidFlow/wells/WellControls.hpp"
+
+namespace geosx
+{
+
+using namespace dataRepository;
+using namespace constitutive;
+
+namespace
+{
+
+// This is meant to be specialized to work, see below
+template< typename COMPOSITIONAL_RESERVOIR_SOLVER > class
+  CompositionalCatalogNames {};
+
+// Class specialization for a RESERVOIR_SOLVER set to CompositionalMultiphaseFlow
+template<> class CompositionalCatalogNames< CompositionalMultiphaseBase >
+{
+public:
+  static string name() { return "CompositionalMultiphaseReservoir"; }
+};
+/*
+   // Class specialization for a RESERVOIR_SOLVER set to MultiphasePoromechanics
+   template<> class CompositionalCatalogNames< MultiphasePoromechanics >
+   {
+   public:
+   static string name() { return "CompositionalMultiphasePoromechanicsReservoir"; }
+   };
+ */
+
+}
```

Comment:
Yes, please, this would simplify a lot the grouping I do in the GEOSX GUI by solver type. Right now we don't have strict rules about the names of the solvers so coming up with regexp to group the solvers was pretty hard.

Reply:
Yes, I am on board with the suggested changes (revamping the naming scheme), I propose to do the changes after discussion with a larger group possibly in a separate PR to make sure I do them right